src/mutexqueue.h

```diff
@@ -0,0 +1,35 @@
+#ifndef __MUTEXQUEUE_H
+#define __MUTEXQUEUE_H
+
+#include <stdbool.h>
+#include "fifo.h"
+
+/*
+ * A thread-safe queue, protected by a mutex.
+ *
+ * Supports:
+ *   - Adding an item to the end of the queue
+ *   - Adding a list of items (Fifo) to the end of the queue
+ *   - Insertion of a priority item at the beginning of the queue (but after existing priority items)
+ *   - Removing an item from the beginning of the queue
+ *   - Removing ALL items as (as new Fifo) from the queue
+ *   - Synchronous waiting on the queue for new items
+ *
+ * The caller is responsible for memory management for items in the queue.
+ */
+
+
+/* The mutexQueue is an opaque structure.  */
+typedef void * mutexQueue;
```

Comment:
Two issues here.
1. At the time I wrote this, I followed the `sds` paradigm which creates a type that represents a pointer.  I've since come to think of this as an anti-pattern which just unnecessarily obfuscates the pointer.  
2. Using a typedef with void doesn't help type safety as it actually resolves to untyped void.  Instead, we can type this to a struct (even if the struct is unknown) - like it's done in fifo.

```suggestion
typedef struct mutexQueue mutexQueue;
```

Note: changing the type to not include the pointer will require a lot of minor code updates.